src/resources/views/crud/inc/form_fields_script.blade.php

```diff
@@ -0,0 +1,156 @@
+     * too, by exposing the main components (name, wrapper, input).
+     */
+    class CrudField {
+        constructor(name) {
+            this.name = name;
+            this.wrapper = $(`[bp-field-name*="${name}"][bp-field-wrapper]`).first();
+
+            if (this.wrapper.length === 0) {
+                console.error(`CrudField error! Could not select WRAPPER for "${this.name}"`);
+            }
+
+            this.type = this.wrapper.attr('bp-field-type');
+            this.$input = this.mainInput;
+            this.input = this.$input[0];
+
+            // Validate that the field has been found
+            if(!this.input) {
+                console.error(`CrudField error! Could not select INPUT for "${this.name}"`);
+            }
+        }
+
+        get value() {
+            let value = this.input.value;
+
+            // Parse the value if it's a number
+            if (value.length && !isNaN(value)) {
+                value = Number(value);
+            }
+
+            return value;
+        }
+
+        get mainInput() {
+            let input;
+
+            // search input in ancestors
+            input = this.wrapper.closest('[bp-field-main-input]');
+
+            // search input in children
+            if (input.length === 0) {
+                input = this.wrapper.find('[bp-field-main-input]');
+            }
+
+            // if no bp-field-main-input has been declared in the field itself, try to find an input with that name inside wraper
+            if (input.length === 0) {
+                input = this.wrapper.find(`input[bp-field-name="${this.name}"], textarea[bp-field-name="${this.name}"], select[bp-field-name="${this.name}"]`).first();
+            }
+
+            // if nothing works, use the first input found in field wrapper.
+            if(input.length === 0) {
+                input = this.wrapper.find('input, textarea, select').first();
+            }
+
+            return input;
+        }
+
+        change(closure) {
+            const bindedClosure = closure.bind(this);
+            const fieldChanged = (event, values) => bindedClosure(this, event, values);
+
+            if(this.isSubfield) {
+                window.crud.subfieldsCallbacks = window.crud.subfieldsCallbacks ?? [];
+                window.crud.subfieldsCallbacks[this.subfieldHolder] = window.crud.subfieldsCallbacks[this.subfieldHolder] ?? [];
+                if(!window.crud.subfieldsCallbacks[this.subfieldHolder].some( callbacks => callbacks['fieldName'] === this.name )) {
+                    window.crud.subfieldsCallbacks[this.subfieldHolder].push({fieldName: this.name, closure: closure, field: this});
+                }
+                return this;
+            }
+
+            this.input?.addEventListener('input', fieldChanged, false);
+            this.$input.change(fieldChanged);
+            fieldChanged();
+
+            return this;
+        }
+
+        onChange(closure) {
+            this.change(closure);
+        }
```

Comment:
If we do not `return` in `onChange` then:
- the return of `change` will be a CrudField
- the return of `onChange` will be undefined

In order to make them interchangeable, we need to return the same thing:

```suggestion
        onChange(closure) {
            return this.change(closure);
        }
```